### PR TITLE
Issue plugin

### PR DESCRIPTION
**Pull request by JayPHP**

### CODE REVIEW

File: config/config.sample.yml [hunk 1]
```diff
@@ -88,3 +88,6 @@ web-api:
 dev-mode:
   enable: true
 
+github:
+  username: %username%
+  password: %password%
```
Comment (JayPHP) [inline config/config.sample.yml]: Will add new line

File: src/Chat/Entities/MainSiteUser.php [hunk 2]
```diff
@@ -43,4 +51,9 @@ public function getTwitterHandle()
     {
         return $this->twitterHandle;
     }
+
+    public function getGithubUsername()
+    {
+        return $this->githubUsername;
+    }
 }
```
Comment (JayPHP) [inline src/Chat/Entities/MainSiteUser.php]: Will add new line

File: src/Plugins/Issue.php [hunk 3]
```diff
@@ -0,0 +1,169 @@
+<?php declare(strict_types=1);
+
+// TODO - Scrape for GH profiles and ping detect
```
Comment (JayPHP) [inline src/Plugins/Issue.php]: Need to remove this

File: src/Plugins/Issue.php [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            $body = yield from $this->getTextFromCommand($content[1], $command);
+        }
+
+        yield from $this->createIssue($title, $body, $command->getId());
+
+        return $this->chatClient->postReply($command, $this->response);
+    }
+
+    private function createIssue(string $title, $body = '', int $id)
+    { // TODO Handle different status returns
```
Comment (JayPHP) [inline src/Plugins/Issue.php]: As discussed, we're happy with just not working / working for now. Though I could add some logging to make it easier to debug

File: src/Chat/Entities/MainSiteUser.php (hunk 2 above)
Comment (PeeHaa) [inline src/Chat/Entities/MainSiteUser.php]: Can you add a new line please?

**Comment by JayPHP** [comment]
@PeeHaa Could you pull again and test 2fa please? If working, should be ready to merge :)

File: src/Plugins/Issue.php [hunk 5]
```diff
@@ -17,10 +15,10 @@
 class Issue extends BasePlugin
 {
     const USAGE = 'Usage: `!!issue [<title> - <body>]`';
-    const ISSUE_URL = 'https://api.github.com/repos/Room-11/Jeeves/issues';
+    const ISSUE_URL = 'https://api.github.com/repos/JayPHP/Project-Template/issues';
```
Comment (kelunik) [inline src/Plugins/Issue.php]: Why did this URL change?

File: src/Plugins/Issue.php [hunk 6]
```diff
@@ -17,10 +15,10 @@
...
     const CHAT_URL_EXP = '~^http://chat\.stackoverflow\.com/transcript/message/(\d+)(#\d+)?$~';
     const PING_EXP = '/@([^\s]+)(?=$|\s)/';
-    const REQUIRED_AUTH = ['username', 'password'];
+    const REQUIRED_AUTH = ['username', 'password', 'token'];
```
Comment (kelunik) [inline src/Plugins/Issue.php]: Why is the `token` required?

Comment (kelunik) [inline src/Plugins/Issue.php]: If the value is not required, then the key should not be required.

**Pull request merged**